mobile/android/app/src/main/java/ru/spb/arcadia/product/assistant/android/AuthenticationSessionModule.java

```diff
@@ -0,0 +1,124 @@
+package ru.spb.arcadia.product.assistant.android;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.support.customtabs.CustomTabsIntent;
+
+import com.facebook.react.bridge.ActivityEventListener;
+import com.facebook.react.bridge.BaseActivityEventListener;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+/**
+ * Created by alexander.shevnin on 31/03/2018.
+ */
+
+public class AuthenticationSessionModule extends ReactContextBaseJavaModule implements LifecycleEventListener {
+
+    private Promise currentSession;
+    private Uri requestedRedirectUri;
+
+    private void cleanUp() {
+        this.currentSession = null;
+        this.requestedRedirectUri = null;
+    }
+
+    private ActivityEventListener mEventListener = new BaseActivityEventListener(){
+        @Override
+        public void onNewIntent(Intent intent) {
+            super.onNewIntent(intent);
+            tryToProcessLogin(intent);
+        }
+    };
+
+
+    public AuthenticationSessionModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+
+        reactContext.addLifecycleEventListener(this);
+        reactContext.addActivityEventListener(this.mEventListener);
```

Comment:
should we unsubscribe?

Reply:
I *was* looking for a way to do it but I couldn't find anything. The best idea I have is to do it in onHostDestroy

Reply:
and this is a singleton so we're probably alright...